### PR TITLE
BUG: No affect both index and column(GH 13522)

### LINKED ISSUE #13522: Assigning a string index as a column once can make further assignments affect both index and column.
If you assign the dataframe's index to as a column as well, without resetting the index, any subsequent assignments on the column seem to affect both the column as well as the index.
The behaviour is there for Index, Int64Index, and Float64Index but not DatetimeIndex.

#### Code Sample, a copy-pastable example if possible.
```python
>>> import pandas as pd
>>> df=pd.DataFrame(index=['A', 'B', 'C'])
>>> df['X']=df.index
>>> df
   X
A  A
B  B
C  C
>>> df['X']=['x', 'y', 'z']
>>> df
   X
x  x
y  y
z  z
>>> df.ix[0, 'X']='q'
>>> df
   X
q  q
y  y
z  z
```
#### Expected Output
```python
   X
A  x
B  y
C  z
```
#### output of ``pd.show_versions()``
```python
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.11.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.1
nose: 1.3.7
pip: 8.1.2
setuptools: 23.0.0
Cython: 0.24
numpy: 1.10.4
scipy: 0.17.1
statsmodels: None
xarray: 0.7.2
IPython: None
sphinx: None
patsy: None
dateutil: 2.4.1
pytz: 2016.4
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.5.2
matplotlib: 1.5.1
openpyxl: 2.3.2
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.13
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.40.0
pandas_datareader: None
```

### PR DESCRIPTION
**Pull request by conquistador1492**
- [x] closes #13522 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
can you run an asv and see if anything changed signficantly. lgtm otherwise.

File: doc/source/whatsnew/v0.19.0.txt [hunk 1]
```diff
@@ -861,6 +861,7 @@ Bug Fixes
 - Bug in ``io.json.json_normalize()``, where non-ascii keys raised an exception (:issue:`13213`)
 - Bug when passing a not-default-indexed ``Series`` as ``xerr`` or ``yerr`` in ``.plot()`` (:issue:`11858`)
 - Bug in area plot draws legend incorrectly if subplot is enabled or legend is moved after plot (matplotlib 1.5.0 is required to draw area plot legend properly) (issue:`9161`, :issue:`13544`)
+- Bug in ``DataFrame``, assigning a string index as a column once can make further assignments affect both index and column (:issue:`13522`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: maybe something like: Bug in ``DataFrame`` assignment with an object-dtyped ``Index`` where the resultant column is mutable to the original object.

**Comment by conquistador1492** [comment]
@jreback all benchmarks not significantly changed beside groupby.groupby_frame_singlekey_integer.time_sum. It ranges from 1 to 1.2

**Comment by jreback** [comment]
thanks!

**Pull request merged**